packages/flutter/test/gestures/drag_test.dart

```diff
@@ -648,6 +649,601 @@ void main() {
+
+    // _moveDeltaBeforeFrame = { }
+
+    log.add('-g');
+    // #6 pointer move to down 10.0, received delta should be (0, 10.0).
+    tester.route(pointer6.move(const Offset(0, 120.0)));
+
+    // _moveDeltaBeforeFrame = {
+    //   6: Offset(0, 10),
+    // }
+
+    log.add('-h');
+    // #8 pointer move to up 10, received delta should be (0, -10.0).
+    tester.route(pointer8.move(const Offset(0, -120.0)));
+
+    // _moveDeltaBeforeFrame = {
+    //   6: Offset(0, 10),
+    //   8: Offset(0, -10),
+    // }
+
+    log.add('-i');
+    // #5 pointer move to down 10.0, received delta should be (0.0, 0.0).
+    tester.route(pointer5.move(const Offset(0, 130.0)));
+
+    // _moveDeltaBeforeFrame = {
+    //   5: Offset(0, 10),
+    //   6: Offset(0, 10),
+    //   8: Offset(0, -10),
+    // }
+
+    log.add('-j');
+    // #7 pointer move to up 10, received delta should be (0.0, 0.0).
+    tester.route(pointer7.move(const Offset(0.0, -130.0)));
+
+    tester.route(pointer5.up());
+    tester.route(pointer6.up());
+    tester.route(pointer7.up());
+    tester.route(pointer8.up());
+
+    // Tear down 'currentSystemFrameTimeStamp'
+    SchedulerBinding.instance.handleBeginFrame(Duration.zero);
+    SchedulerBinding.instance.handleDrawFrame();
+
+    expect(log, <String>[
+      '-a',
+      'drag-update (Offset(0.0, 100.0))',
+      '-b',
+      'drag-update (Offset(0.0, 10.0))',
+      '-c',
+      'drag-update (Offset(0.0, -100.0))',
+      '-d',
+      'drag-update (Offset(0.0, -10.0))',
+      '-e',
+      'drag-update (Offset(0.0, 10.0))',
+      '-f',
+      'drag-update (Offset(0.0, -10.0))',
+      '-g',
+      'drag-update (Offset(0.0, 10.0))',
+      '-h',
+      'drag-update (Offset(0.0, -10.0))',
+      '-i',
+      'drag-update (Offset(0.0, 0.0))',
+      '-j',
+      'drag-update (Offset(0.0, 0.0))'
+    ]);
+  });
+
+  testGesture('Pan drag with multiple pointers - maxAllPointers', (GestureTester tester) {
+    final PanGestureRecognizer drag =
+    PanGestureRecognizer()
+      ..multitouchDragStrategy = MultitouchDragStrategy.maxAllPointers;
+
+    final List<String> log = <String>[];
+    drag.onUpdate = (DragUpdateDetails details) { log.add('drag-update (${details.delta})'); };
+
+    final TestPointer pointer5 = TestPointer(5);
+    final PointerDownEvent down5 = pointer5.down(Offset.zero);
+    drag.addPointer(down5);
+    tester.closeArena(5);
+    tester.route(down5);
+
+    log.add('-a');
+    // #5 pointer move (100.0, 100.0), received delta should be (100.0, 100.0).
+    tester.route(pointer5.move(const Offset(100.0, 100.0)));
+
+    // _moveDeltaBeforeFrame = { 5: Offset(100, 100), }
+
+    // Put down the second pointer 6.
+    final TestPointer pointer6 = TestPointer(6);
+    final PointerDownEvent down6 = pointer6.down(Offset.zero);
+    drag.addPointer(down6);
+    tester.closeArena(6);
+    tester.route(down6);
+
+    log.add('-b');
+    // #6 pointer move (110.0, 110.0), received delta should be (10, 10).
```

Comment:
This is not the same as the panning algorithm I described in https://github.com/flutter/flutter/issues/38926, although I might have been wrong. Can you verify the native behavior of panning on iOS and see if it only checks the border pointers or if it averages all pointers? Also, what if the border pointers for x axis are different from those for y axis, for example:
```
5: Offset(-100, 0),
6: Offset(100, 0),
7: Offset(0, -100),
8: Offset(0, 100),
```
what's the behavior on native for cases like this?

Reply:
I test the `UIPanGestureRecognizer` for iOS and indeed like you said, it takes the average of all fingers.
Here is the test code,
```dart
//
//  ViewController.swift
//  panning
//
//  Created by 徐宝林(Ava) on 2024/2/18.
//

import UIKit

class ViewController: UIViewController {
    
    private let pannableView: UIView = {
        // Initialize View
        let view = UIView(frame: CGRect(origin: .zero,
                                        size: CGSize(width: 300.0, height: 300.0)))

        // Configure View
        view.backgroundColor = .blue
        view.translatesAutoresizingMaskIntoConstraints = false

        return view
    }()
    
    @objc private func didPan(_ sender: UIPanGestureRecognizer) {
//        pannableView.center = sender.location(in: view)
        let center = pannableView.center
        let trans = sender.translation(in: view)
        pannableView.center = CGPointMake(center.x + trans.x, center.y + trans.y)
        sender.setTranslation(CGPointZero, in: view)
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        // Add to View Hierarchy
        view.addSubview(pannableView)

        // Center Pannable View
        pannableView.center = view.center

        // Initialize Swipe Gesture Recognizer
        let panGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(didPan(_:)))

        // Add Swipe Gesture Recognizer
        pannableView.addGestureRecognizer(panGestureRecognizer)
    }
}
```

Reply:
> 
> ```
> 5: Offset(-100, 0),
> 6: Offset(100, 0),
> 7: Offset(0, -100),
> 8: Offset(0, 100),
> ```
> 
> what's the behavior on native for cases like this?

The draggable object will remain stationary.